x/programs/examples/counter-external/src/lib.rs

```diff
@@ -14,3 +14,22 @@ pub fn get_value(ctx: &mut Context, external: Address, of: Address) -> u64 {
     let ctx = ctx.to_extern(external, 1_000_000, 0);
     counter::get_value(&ctx, of)
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::*;
+
+    #[test]
+    fn get_value_exists() {
+        let mut ctx = Context::new();
+        let external = Address::new([0; 33]);
+        let of = Address::new([1; 33]);
+
+        // value `of` has in the external contract
+        let value = 5_u64;
+        ctx.mock_call_function(external, "get_value", of, 1_000_000, 0, value);
```

Comment:
```suggestion
        ctx.mock_function_call(external, "get_value", of, 1_000_000, 0, value);
```

Reply:
Can we somehow use [`mockall`](https://docs.rs/mockall/latest/mockall/) under the hood?
The syntax is much more intuitive
```rust
mock.expect_foo()
    .with(predicate::eq(4))
    .times(1)
    .returning(|x| x + 1);
```